### ISSUE TITLE
Improve suggestions when minikube create runs into error

### ISSUE THREAD

**Issue by tejal29** [kubernetes/minikube#9273]
hey folks, i was trying `gcloud alpha dev code` on one for the cloud code sample. 

```
➜  go-cloud-run-hello-world git:(master) ✗ gcloud alpha code dev
This action requires the installation of components: [minikube]
Your current Cloud SDK version is: 302.0.0
Installing components from version: 302.0.0
┌───────────────────────────────────────────────────────────────────┐
│                These components will be installed.                │
├──────────────────────┬─────────────────────┬──────────────────────┤
│         Name         │       Version       │         Size         │
├──────────────────────┼─────────────────────┼──────────────────────┤
│ Minikube             │              1.12.0 │             23.1 MiB │
│ Minikube             │                     │                      │
└──────────────────────┴─────────────────────┴──────────────────────┘
For the latest full release notes, please visit:
  https://cloud.google.com/sdk/release_notes
Do you want to continue (Y/n)?  Y
╔════════════════════════════════════════════════════════════╗
╠═ Creating update staging area                             ═╣
╠════════════════════════════════════════════════════════════╣
╠═ Installing: Minikube                                     ═╣
╠════════════════════════════════════════════════════════════╣
╠═ Installing: Minikube                                     ═╣
╠════════════════════════════════════════════════════════════╣
╠═ Creating backup and activating new installation          ═╣
╚════════════════════════════════════════════════════════════╝
Performing post processing steps...done.                                                                                            
Update done!
Restarting command:
  $ gcloud alpha code dev
Starting development environment 'gcloud-local-dev' ...
ERROR: (gcloud.alpha.code.dev) Command '[u'/Users/tejaldesai/Downloads/google-cloud-sdk2/bin/minikube', u'start', u'-p', u'gcloud-local-dev', u'--keep-context', u'--interactive=false', u'--delete-on-failure', u'--install-addons=false', u'--vm-driver=docker', u'--container-runtime=docker']' returned non-zero exit status 70
```

I ran into exit status 70 error.  This did not give me enough information on how to recover. 
I decided to run the command myself to see the exact error after getting rid of all the "`"

I ran into this error
```
/Users/tejaldesai/Downloads/google-cloud-sdk2/bin/minikube start -p gcloud-local-dev --keep-context --interactive=false --delete-on-failure --install-addons=false --vm-driver=docker --container-runtime=docker
😄  [gcloud-local-dev] minikube v1.12.0 on Darwin 10.15.3
✨  Using the docker driver based on existing profile
👍  Starting control plane node gcloud-local-dev in cluster gcloud-local-dev
🔄  Restarting existing docker container for "gcloud-local-dev" ...
🤦  StartHost failed, but will try again: provision: Error getting config for native Go SSH: open /Users/tejaldesai/.minikube/machines/gcloud-local-dev/id_rsa: no such file or directory
✋  Stopping "gcloud-local-dev" in docker ...
🔥  Deleting "gcloud-local-dev" in docker ...
🔥  Creating docker container (CPUs=2, Memory=7916MB) ...
😿  Failed to start docker container. "minikube start -p gcloud-local-dev" may fix it: creating host: create: creating: create kic node: container name "gcloud-local-dev": container exited unexpectedly

💥  The minikube docker container exited unexpectedly.
💡  If you are still interested to make docker driver work. The following suggestions might help you get passed this issue:

	- Prune unused docker images, volumes and abandoned containers.

	- Restart your docker service

	- Ensure your docker daemon has access to enough CPU/memory resources. 

	- Docs https://docs.docker.com/docker-for-mac/#resources

	- Delete and recreate minikube cluster
		minikube delete
		minikube start --driver=docker

💣  error provisioning host: Failed to start host: creating host: create: creating: create kic node: container name "gcloud-local-dev": container exited unexpectedly

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

I tried the solutions mentioned in the suggestions.
1. 
```
➜  go-cloud-run-hello-world git:(master) ✗ docker system prune
WARNING! This will remove:
  - all stopped containers
  - all networks not used by at least one container
  - all dangling images
  - all dangling build cache

Are you sure you want to continue? [y/N] y
Deleted Containers:
1d8afdb71bb7e8efead75dd771c8ae6f00d1616402091a968d184f1133c2def5

Deleted Images:
deleted: sha256:436abe05a8f0987b3854573eeb70ceb94109144484b8fd781580a28d74eb7716
deleted: sha256:7ec2dd2ab91ebfaad0f4220deb70dbc19c36d701dcbcdcdb6de42b7d148a1897
deleted: sha256:a0ca127ea1abe1e220704d57efed9dd1f29c2f438c86bac0503431ce33451d31
deleted: sha256:ee49ca918280003fbf07ba8c06f23e61a1574f8271e736de3508bec7817c4fb1
deleted: sha256:fbab32a8f18c461775b322d639a28993bfa941b8d3d43bbde07fa2bc7a762ff4
deleted: sha256:42413972a6dc985ec5c4f8fdd8cae47f5ad01d8dab6daf95dbb9b7309aedf9e2
deleted: sha256:eab6071800a7dd8e0b0d091d978c3df0c6fb3ccf9a5fdc76e43f9c2d36710071
deleted: sha256:450006f80fd97fcc6a734526bd2161499610ec24fd248edd378e4be6869808f6

Total reclaimed space: 14.95MB
```
This did not help. 

2. Restarting docker did not help either.

3. Made sure docker daemon had enough resources. 
![image](https://user-images.githubusercontent.com/1534539/93632757-ba124880-f9a2-11ea-8cde-5aca2c25e782.png)

4. Ran `minikube delete` I ran into this error
```
/Users/tejaldesai/Downloads/google-cloud-sdk2/bin/minikube start -p gcloud-local-dev --keep-context --interactive=false --delete-on-failure --install-addons=false --vm-driver=docker --container-runtime=docker
😄  [gcloud-local-dev] minikube v1.12.0 on Darwin 10.15.3
✨  Using the docker driver based on existing profile
👍  Starting control plane node gcloud-local-dev in cluster gcloud-local-dev
🔄  Restarting existing docker container for "gcloud-local-dev" ...
🤦  StartHost failed, but will try again: provision: get ssh host-port: get port 22 for "gcloud-local-dev": docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" gcloud-local-dev: exit status 1
stdout:


stderr:
Template parsing error: template: :1:4: executing "" at <index (index .NetworkSettings.Ports "22/tcp") 0>: error calling index: index of untyped nil

✋  Stopping "gcloud-local-dev" in docker ...
🔥  Deleting "gcloud-local-dev" in docker ...
🔥  Creating docker container (CPUs=2, Memory=7916MB) ...
😿  Failed to start docker container. "minikube start -p gcloud-local-dev" may fix it: creating host: create: creating: create kic node: container name "gcloud-local-dev": container exited unexpectedly

💥  The minikube docker container exited unexpectedly.
💡  If you are still interested to make docker driver work. The following suggestions might help you get passed this issue:

	- Prune unused docker images, volumes and abandoned containers.

	- Restart your docker service

	- Ensure your docker daemon has access to enough CPU/memory resources. 

	- Docs https://docs.docker.com/docker-for-mac/#resources

	- Delete and recreate minikube cluster
		minikube delete
		minikube start --driver=docker

💣  error provisioning host: Failed to start host: creating host: create: creating: create kic node: container name "gcloud-local-dev": container exited unexpectedly

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

I realized, i should have added `-p` option. (This should have been suggested. Its easy to refer the command line args and spit the right command line options for delete)

5. Ran ` minikube delete -p gcloud-local-dev` and this did the magic
```
➜  go-cloud-run-hello-world git:(master) ✗ /Users/tejaldesai/Downloads/google-cloud-sdk2/bin/minikube start -p gcloud-local-dev --keep-context --interactive=false --delete-on-failure --install-addons=false --vm-driver=docker --container-runtime=docker
😄  [gcloud-local-dev] minikube v1.12.0 on Darwin 10.15.3
❗  Both driver=docker and vm-driver=docker have been set.

    Since vm-driver is deprecated, minikube will default to driver=docker.

    If vm-driver is set in the global config, please run "minikube config unset vm-driver" to resolve this warning.
			
✨  Using the docker driver based on user configuration
👍  Starting control plane node gcloud-local-dev in cluster gcloud-local-dev
🔥  Creating docker container (CPUs=2, Memory=7916MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
🔎  Verifying Kubernetes components...
💗  To connect to this cluster, use: kubectl --context=gcloud-local-dev

❗  /usr/local/bin/kubectl is version 1.15.3, which may be incompatible with Kubernetes 1.18.3.
💡  You can also use 'minikube kubectl -- get pods' to invoke a matching version
```

Expected Results, 
1. looks like `minikube delete -p <>` helped me recover. Can we add that as the first suggestion?

**Comment by medyagh** [comment]
This seems to be a bug that delete on failure should had worked "--delete-on-failure"

**Comment by tejal29** [comment]
Could be possible. There could be chance that freeing up docker was the right solution. 
However, when going through the start code i saw,
If minikube start fails, minikube deletes the profile and atempts to try again.
```
✋  Stopping "gcloud-local-dev" in docker ...
🔥  Deleting "gcloud-local-dev" in docker ...
```

If freeing up docker memory was right solution, then second time running "minikube start" should have worked. 
Another possibility is `deleteMinilubeProfile` has some bug.

**Comment by tstromberg** [comment]
I think this was fixed in v1.13?

**Comment by tstromberg** [comment]
Closing as obsolete, since this was for an older version of minikube (4 releases ago), and we've addressed these issues since.
